source/code/go/src/plugins/oms.go

```diff
@@ -262,6 +326,235 @@ func convert(in interface{}) (float64, bool) {
+						ContainerId:    containerID,
+						FirstOccurance: eventFirstOccurance,
+						LastOccurance:  eventTimeStamp,
+						Count:          eventCount + 1,
+					}
+				} else {
+					PromScrapeErrorEvent[splitString] = KubeMonAgentEventTags{
+						PodName:        podName,
+						ContainerId:    containerID,
+						FirstOccurance: eventTimeStamp,
+						LastOccurance:  eventTimeStamp,
+						Count:          1,
+					}
+				}
+			}
+		}
+	}
+	EventHashUpdateMutex.Unlock()
+	Log("Updating config event hash - Unlocked after update \n ")
+}
+
+// Function to get config error log records after iterating through the two hashes
+func flushKubeMonAgentEventRecords() {
+	for ; true; <-KubeMonAgentConfigEventsSendTicker.C {
+		if skipKubeMonEventsFlush != true {
+			Log("In flushConfigErrorRecords\n")
+			var retries = 2
+			var flushSuccessful = false
+			var resp *http.Response
+			var postError error
+			var elapsed time.Duration
+			var laKubeMonAgentEventsRecords []laKubeMonAgentEvents
+			telemetryDimensions := make(map[string]string)
+			start := time.Now()
+
+			telemetryDimensions["ConfigErrorEventCount"] = strconv.Itoa(len(ConfigErrorEvent))
+			telemetryDimensions["PromScrapeErrorEventCount"] = strconv.Itoa(len(PromScrapeErrorEvent))
+
+			if (len(ConfigErrorEvent) > 0) || (len(PromScrapeErrorEvent) > 0) {
+				EventHashUpdateMutex.Lock()
+				for k, v := range ConfigErrorEvent {
+					tagJson, err := json.Marshal(v)
+
+					if err != nil {
+						message := fmt.Sprintf("Error while Marshalling config error event tags: %s", err.Error())
+						Log(message)
+						SendException(message)
+					} else {
+						laKubeMonAgentEventsRecord := laKubeMonAgentEvents{
+							Computer:       Computer,
+							CollectionTime: start.Format(time.RFC3339),
+							Category:       ConfigErrorEventCategory,
+							Level:          KubeMonAgentEventError,
+							ClusterId:      ResourceID,
+							ClusterName:    ResourceName,
+							Message:        k,
+							Tags:           fmt.Sprintf("%s", tagJson),
+						}
+						laKubeMonAgentEventsRecords = append(laKubeMonAgentEventsRecords, laKubeMonAgentEventsRecord)
+					}
+				}
+
+				for k, v := range PromScrapeErrorEvent {
+					tagJson, err := json.Marshal(v)
+					if err != nil {
+						message := fmt.Sprintf("Error while Marshalling prom scrape error event tags: %s", err.Error())
+						Log(message)
+						SendException(message)
+					} else {
+						laKubeMonAgentEventsRecord := laKubeMonAgentEvents{
+							Computer:       Computer,
+							CollectionTime: start.Format(time.RFC3339),
+							Category:       PromScrapingErrorEventCategory,
+							Level:          KubeMonAgentEventWarning,
+							ClusterId:      ResourceID,
+							ClusterName:    ResourceName,
+							Message:        k,
+							Tags:           fmt.Sprintf("%s", tagJson),
+						}
+						laKubeMonAgentEventsRecords = append(laKubeMonAgentEventsRecords, laKubeMonAgentEventsRecord)
+					}
+				}
+				EventHashUpdateMutex.Unlock()
```

Comment:
please add a log with methodname while locking & unlocking to we can use logs to find deadlocks.